Goals Capture: Handle the `create-blog-lp` ref parameter

ivan-ottinger opened the pull request:
#### Proposed Changes

This PR builds on top of https://github.com/Automattic/wp-calypso/pull/64145 and automatically selects the "Write" goal if the URL parameter `ref` equals to `create-blog-lp`.

|  Before | After |
| ------------- | ------------- |
| ![Markup on 2022-06-16 at 16:08:12](https://user-images.githubusercontent.com/25105483/174088600-0e068b32-175a-4319-a855-7e91fbb3f66e.png) | ![Markup on 2022-06-16 at 16:07:26](https://user-images.githubusercontent.com/25105483/174088607-0f07a476-f294-4cb2-8108-bef89041adbb.png) |

#### Testing Instructions

1. Go to `calypso.localhost:3000/setup/goals?siteSlug=[site_address]&flags=signup/goals-step&ref=create-blog-lp` where `[site_address]` is the address of the site in form of `something.wordpress.com`.
2. When the Goals Capture step displays, the "Write" goal should be selected automatically.
3. Explore the state through Redux DevTools → the "Write" goal should be selected there as well.
4. Click through other goals - they should get selected as expected.
5. Try to pass incorrect `ref` parameter value, or omit the `ref` parameter altogether. → The Goals step should render correctly without having any goal selected.

ivan-ottinger committed:
Initial commit

yansern commented on client/landing/stepper/declarative-flow/internals/steps-repository/goals/select-goals.tsx:
```diff
@@ -18,6 +20,18 @@ export const SelectGoals: React.FC< SelectGoalsProps > = ( {
 	const translate = useTranslate();
 	const goalOptions = useGoals();
 
+	const comingFromCreateBlogLanding = getQueryArgs()?.ref === 'create-blog-lp';
```
Just for future-proofing, what do you think about putting this into some referral goals mapping.
```
const someReferralGoals = {
    'create-blog-lp': [ SiteGoal.Write ],
    'something-in-the-future': [ SiteGoal.Sell, SiteGoal.Promote ]
}
```

ivan-ottinger replied:
There's only `create-blog-lp` ref available currently, right? 🤔 

Is there any plan to have other refs in place already? Or I guess there's high chance we might need other refs in the future?

yansern replied:
Yup, there might be one for build and sell, e.g. https://wordpress.com/website-builder/ this one is for build.

But not something to worry for now. Future-proofing helps turn this into a one-line effort in the future.

ivan-ottinger replied:
Thanks! This is now implemented in the latest commit https://github.com/Automattic/wp-calypso/pull/64701/commits/9743e3bdd324d707febb87085f60915cb2f3fd92.

yansern commented on client/landing/stepper/declarative-flow/internals/steps-repository/goals/index.tsx:
```diff
@@ -76,6 +80,19 @@ const GoalsStep: Step = ( { navigation } ) => {
+	//const goals = comingFromCreateBlogLanding
+	//	? [ Onboard.SiteGoal.Write ]
+	//	: useSelect( ( select ) => select( ONBOARD_STORE ).getGoals() );
+
```
I think it's a good question. How do you decide between existing selected goals vs goals to be automatically selected from referral ID?

I think it should be something like this:
- if there are no existing selected goals AND there is a referral ID, automatically select goals.
- if there are existing selected goals (persisted from the data store), ignore referral ID (if they exist), and don't automatically select goals.

ivan-ottinger committed:
Add URL `ref` parameter map object

The commit introduces `refGoals` object that will simplify adding extra `ref` URL parameter values in the future.

mashikag reviewed (approved):
I tested it locally and it works as expected. 👍 
![Screenshot 2022-06-21 at 17 38 57](https://user-images.githubusercontent.com/2019970/174841577-818c8e84-3bb7-4662-b77e-9dd4d0cca964.png)

ivan-ottinger commented:
Thank you for the merge Yan!
